src/v1/account_validity.rs

```diff
@@ -0,0 +1,60 @@
+//! [GET /_synapse/admin/v1/account_validity/validity](https://github.com/matrix-org/synapse/blob/master/docs/admin_api/account_validity.rst)
+
+use std::time::SystemTime;
+
+use ruma::api::ruma_api;
+use ruma::identifiers::UserId;
+
+ruma_api! {
+    metadata: {
+        description: "Extend the account validity.",
+        method: POST,
+        name: "set_account_validity",
+        path: "/_synapse/admin/v1/account_validity/validity",
+        rate_limited: false,
+        authentication: None, // AccessToken?
+    }
+
+    request: {
+        /// user ID for the account to renew
+        pub user_id: &'a UserId,
+
+        /// This is an optional parameter, it overrides the expiration date,
+        /// which otherwise defaults to now + validity period configured at the server
+        #[serde(
+            with = "ruma_serde::time::opt_ms_since_unix_epoch",
+            skip_serializing_if = "Option::is_none"
+        )]
+        pub expiration_ts: Option<SystemTime>,
+
+        /// This is an optional parameter, it enables/disables sending renewal emails to the user.
+        /// Defaults to true.
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub enable_renewal_emails: Option<bool>,
```

Comment:
This can just be a bool, using `serde(deserialize_with = "ruma::serde::default_true")`.

Reply:
done

Reply:
It won't. Since `None` is semantically the same thing as `Some(true)` here, there's not really a need to encode them differently. I also forgot: You can then use the `skip_serializing_if` attribute with `is_true` (I think that also exists in ruma_serde).